samples/InputEcsSample/main.cpp

```diff
@@ -93,65 +96,67 @@ int main()
-            newComponent.scale = NovelRT::Maths::GeoVector2F::zero();
-            newComponent.scale += scale;
-            newComponent.positionAndLayer += move;
-            transforms.PushComponentUpdateInstruction(entity, newComponent);
-        }
-    });
+            ComponentView<NovelRT::Ecs::Input::InputEventComponent> events =
+                catalogue.template GetComponentView<NovelRT::Ecs::Input::InputEventComponent>();
+            ComponentView<TransformComponent> transforms = catalogue.template GetComponentView<TransformComponent>();
+            NovelRT::Maths::GeoVector2F scale = NovelRT::Maths::GeoVector2F::Zero();
+            NovelRT::Maths::GeoVector2F move = NovelRT::Maths::GeoVector2F::Zero();
+
+            NovelRT::Ecs::Input::InputEventComponent input;
+            if (events.TryGetComponent(up, input) &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{0.0f, 5.0f};
+            }
+
+            if (events.TryGetComponent(down, input) &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{0.0f, -5.0f};
+            }
+
+            if (events.TryGetComponent(left, input) &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{-5.0f, 0.0f};
+            }
+
+            if (events.TryGetComponent(right, input) &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                move += NovelRT::Maths::GeoVector2F{5.0f, 0.0f};
+            }
+
+            if (events.TryGetComponent(buttonA, input) &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                scale += NovelRT::Maths::GeoVector2F{0.2f, 0.2f};
+            }
+
+            if (events.TryGetComponent(buttonB, input) &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                scale += NovelRT::Maths::GeoVector2F{-0.2f, -0.2f};
+            }
+
+            if (events.TryGetComponent(mouseClick, input) &&
+                (input.state == KeyState::KeyDown || input.state == KeyState::KeyDownHeld))
+            {
+                logger.logInfo("Clicked at {}, {}", input.mousePositionX, input.mousePositionY);
+            }
+
+            for (auto [entity, transform] : transforms)
+            {
+                TransformComponent newComponent{};
+                newComponent.scale = NovelRT::Maths::GeoVector2F::Zero();
+                newComponent.scale += scale;
+                newComponent.positionAndLayer += move;
+                transforms.PushComponentUpdateInstruction(entity, newComponent);
+            }
+        });
```

Comment:
????

Reply:
honestly, i have no clue what it did due to the sheer amount